test/unittests/evm_eip2929_test.cpp

```diff
@@ -0,0 +1,258 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2021 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+/// This file contains EVM unit tests for EIP-2929 "Gas cost increases for state access opcodes"
+/// https://eips.ethereum.org/EIPS/eip-2929
+
+#include "evm_fixture.hpp"
+
+using namespace evmc::literals;
+using evmone::test::evm;
+
+TEST_P(evm, eip2929_case1)
+{
+    // https://gist.github.com/holiman/174548cad102096858583c6fbbb0649a#case-1
+    rev = EVMC_BERLIN;
+    msg.sender = 0x0000000000000000000000000000000000000000_address;
+    msg.destination = 0x000000000000000000000000636F6E7472616374_address;
+    const auto code =
+        "0x60013f5060023b506003315060f13f5060f23b5060f3315060f23f5060f33b5060f1315032315030315000";
+
+    execute(code);
+    EXPECT_GAS_USED(EVMC_SUCCESS, 8653);
+    EXPECT_EQ(result.output_size, 0);
+
+    const auto& r = host.recorded_account_accesses;
+    ASSERT_EQ(r.size(), 24);
+    EXPECT_EQ(r[0], msg.sender);
```

Comment:
Maybe could be better-looking with some fancy assert
```
EXPECT_THAT(r, ElementsAre(msg.sender, msg.destination, ...));
```

Reply:
This is fine so I don't need to count these in my head. And I don't need to import Google Mock.